src/Npgsql/TaskExtensions.cs

```diff
@@ -82,18 +81,9 @@ internal static async Task WithCancellation(this Task task, CancellationToken ca
         }
 
         internal static Task<T> WithCancellationAndTimeout<T>(this Task<T> task, NpgsqlTimeout timeout, CancellationToken cancellationToken)
-        {
-            return task
-                .WithCancellation(cancellationToken)
-                .WithTimeout(timeout);
-        }
+            => task.WithCancellation(cancellationToken).WithTimeout(timeout);
```

Comment:
```csharp
[MethodImpl(MethodImplOptions.AggressiveInlining)]
internal static TTask WithCancellationAndTimeout<TTask>(this TTask task, NpgsqlTimeout timeout, CancellationToken cancellationToken)
    where TTask : Task => task.WithCancellation(cancellationToken).WithTimeout(timeout);
```

Reply:
I can add, but:
a) Any reason to assume it isn't done implicitly (very small function body), and
b) This is a very cold path